src/server/cluster/cluster_family.cc

```diff
@@ -697,11 +697,23 @@ void ClusterFamily::DflyClusterFlushSlots(CmdArgList args, SinkReplyBuilder* bui
   return builder->SendOk();
 }
 
-void ClusterFamily::StartSlotMigrations(std::vector<MigrationInfo> migrations) {
-  // Add validating and error processing
-  for (auto& m : migrations) {
-    auto outgoing_migration = CreateOutgoingMigration(std::move(m));
-    outgoing_migration->Start();
+void ClusterFamily::StartNewSlotMigrations(std::shared_ptr<ClusterConfig> new_config) {
```

Comment:
I think you can use here const ClusterConfig& right?  do we need the shared pointer here